sensor/common/compliance/command_handler_impl.go

```diff
@@ -40,12 +39,12 @@ func (c *commandHandlerImpl) Start() error {
 	return nil
 }
 
-func (c *commandHandlerImpl) Stop(err error) {
-	c.stopC.SignalWithError(err)
+func (c *commandHandlerImpl) Stop(_ error) {
+	c.stopper.Client().Stop()
```

Comment:
It looks like we are loosing information about the error passed here. As `c.stopper.Flow().StopWithError(err)` does not fit here, then maybe we need a replacement for sending a command to stop with error?

Reply:
No, we are not loosing information because the error is always `nil`.

It is exactly the same topic as above - https://github.com/stackrox/stackrox/pull/3931#discussion_r1034489073

I recommend you to check with IDE Find Usages what calls this `Stop()` method and whether there's any single place that gets non-`nil` error. I found zero.

Reply:
I see, so the error was a "plan for later" that we never used

Reply:
Probably. Something of this kind, or API was not separated for internal/external (from goroutine point of view) use cases and external functions were just a copy of internal needs. I guess we'd never know unless we go in archeology.